client/txn.go

```diff
@@ -142,12 +143,13 @@ func (txn *Txn) DebugName() string {
 // serializable isolation. The isolation must be set before any operations are
 // performed on the transaction.
 func (txn *Txn) SetIsolation(isolation roachpb.IsolationType) error {
-	if txn.Proto.Isolation != isolation {
-		if txn.Proto.IsInitialized() {
-			return util.Errorf("cannot change the isolation level of a running transaction")
-		}
-		txn.Proto.Isolation = isolation
+	if txn.retrying {
```

Comment:
I think it would be good to keep the equality checks for retrying transactions: on a retry, it is an error to set the isolation or priority to anything but its previous value.